minicluster/src/main/java/org/apache/accumulo/miniclusterImpl/MiniAccumuloClusterImpl.java

```diff
@@ -1094,6 +1098,35 @@ protected ExecutorService getShutdownExecutor() {
     return executor;
   }
 
+  public void stopProcessesWithTimeout(final ServerType type, final List<Process> procs,
+      final long timeout, final TimeUnit unit) {
+
+    final List<Future<Integer>> futures = new ArrayList<>();
+    for (Process proc : procs) {
+      futures.add(executor.submit(() -> {
+        proc.destroy();
+        proc.waitFor(timeout, unit);
+        return proc.exitValue();
+      }));
+    }
+
+    while (!futures.isEmpty()) {
+      Iterator<Future<Integer>> iter = futures.iterator();
+      while (iter.hasNext()) {
+        Future<Integer> f = iter.next();
+        if (f.isDone()) {
+          try {
+            f.get();
+          } catch (ExecutionException | InterruptedException e) {
+            log.warn("{} did not fully stop after 30 seconds", type, e);
+          } finally {
+            iter.remove();
+          }
+        }
+      }
+    }
```

Comment:
This loop could be really tight while waiting for the futures to complete. Also we should use the actual `timeout` and `unit` in the log.warn instead of the hardcoded 30 seconds.